### DISCUSSION TITLE
Synchronization of FULL_TENSOR data

### DISCUSSION THREAD

**Discussion by jritta** [peridigm/peridigm#278] [Q&A]
Dear community,

I want to implement a stress formulation. Therefore I have to synchronize `FULL_TENSOR` data (vector with 9 entries) across the processors.
Can someone give me a hint which is the best way to do so? So far I created a nine dimensional `Epetra_BlockMap` that is based on the one dimensional map. But I realized that I would also have to make many changed at various places in the code (e.g. `Block_Base` class). What do you think, is there a better and smarter way to approach this problem?

Many thanks for all help and hints in advance!

Cheers,
Jonas

**Comment by sahirbutt** [comment]
Hi Jonas,
I had a similar issue in the past, at that time I used three vectors to store the tensorial information and use the existing vector synchronization mechanisms for synching the three vectors across the processors. Check out how the Deformation_Gradient is stored for the BondAssociatedCorrespondenceMaterial model. 
Best,
Sahir

**Reply by jritta** [reply]
Hi Sahir,

Thank you very much for your fast reply and this hint! That is maybe a very good workaround.

Best,
Jonas

**Comment by jritta** [comment]
In general, what do you think (especially @djlittl) would it be worth (also from performance perspective) to try to implement a synchronization for `FULL_TENSOR` data?

Cheers,
J

**Comment by jritta** [comment]
Thank you for your suggestion! I was finally able to implement the synchronization of `FULL_TENSOR` data by following the approach as it is done for the `VECTOR` or `SCALAR` field.

Best,
Jonas
